### ISSUE TITLE
Every OAuth flow we have is broken - Sys.ArgumentNullException: Value cannot be null. Parameter name: conversationId

### ISSUE THREAD

**Issue by shff** [OfficeDev/office-js#2167]
We open dialogs in Outlook using `Office.context.ui.displayDialogAsync` that redirect to an OAuth login page.

After the OAuth flow redirects back to the same HTML page in the extension, calling `Office.onReady()` triggers an internal exception. This happens with no other, just calling `Office.onReady()` triggers it.

All the involved domains are added to `AppDomains` in the manifest xml.

This *does not* happen when the redirect happens fast enough. Extension HTML --> Other website --> Extension HTML. Even when the involved files/URLs are the same, it's fine.

## Your Environment

* Platform [PC desktop, Mac, iOS, Office on the web]: Outlook PC Desktop, Office and Web
* Host [Excel, Word, PowerPoint, etc.]: Outlook
* Office version number: Web, Mac and PC 365 Desktop, latest
* Operating System: All operating systems
* Browser (if using Office on the web): Safari, Firefox, Chrome, Brave, IE11 and Edge

## Expected behavior

Office.js being able to initialize correctly after an OAuth flow redirect back to the plugin.

## Current behavior

We get the following exception when we call `Office.onReady()`:

```
Uncaught Sys.ArgumentNullException: Sys.ArgumentNullException: Value cannot be null.
Parameter name: conversationId
    at Function.Error.create (https://ajax.aspnetcdn.com/ajax/3.5/MicrosoftAjax.js:5:2732)
    at Function.Error.argumentNull (https://ajax.aspnetcdn.com/ajax/3.5/MicrosoftAjax.js:5:3196)
    at Function._validateParameterType (https://ajax.aspnetcdn.com/ajax/3.5/MicrosoftAjax.js:5:1648)
    at Function._validateParameter (https://ajax.aspnetcdn.com/ajax/3.5/MicrosoftAjax.js:5:1080)
    at Function._validateParams (https://ajax.aspnetcdn.com/ajax/3.5/MicrosoftAjax.js:5:629)
    at new Microsoft.Office.Common.ClientEndPoint (https://appsforoffice.microsoft.com/lib/1/hosted/outlook-web-16.01.js:20:98542)
    at Object.connect (https://appsforoffice.microsoft.com/lib/1/hosted/outlook-web-16.01.js:20:108506)
    at OSF.InitializationHelper.setAgaveHostCommunication (https://appsforoffice.microsoft.com/lib/1/hosted/outlook-web-16.01.js:20:130539)
    at s (https://appsforoffice.microsoft.com/lib/1/hosted/Office.js:46:27197)
    at d (https://appsforoffice.microsoft.com/lib/1/hosted/Office.js:46:3609)
```

## Steps to reproduce

1. Open a dialog using `Office.context.ui.displayDialogAsync`
2. Have the dialog redirect to an OAuth flow in another domain included in the Manifest XML
3. Have the OAuth page in step 2 redirect back to the Extension's HTML page
4. The exception above is triggered when calling `Office.onReady()`.

## Link to live example(s)

Office Lab doesn't seem to support Outlook, nor does it seem to support launching dialogs in HTML files like Outlook add-ins do, so it's not really reproducible in it.

We have a live extension published where this happens. [Here](https://appsource.microsoft.com/en-us/product/office/WA200001031). I'm happy to provide test credentials, although creating a fresh account is possible.

## Context

This was working for about two years but it's broken as of today. Our last update was last month.

Our login that uses OAuth is broken. The user has to click twice for it to work.

## Useful logs

The log I added above is all I have.

**Comment by exextoc** [comment]
The above callstack appears to be from Outlook Web Access (OWA), do you have similar exception error for Win32 Outlook as well?

**Comment by exextoc** [comment]
@shff I tried using add by creating a new account. I was not able to login successfully as it required admin to grant some access and flow in dialog got interrupted with different error.

1. Can you please test credentials with which this can be reproduced easily?
2. Did anything change in addin code recently?

**Comment by shff** [comment]
@exextoc It seems to be working fine again. If you were able to login (but got greeted with a permission request) then it probably worked. The previous "workaround" was clicking twice in the "Start with Mailbutler..." button, the first time prompts for the password, the second redirects correctly.

We didn't update the add-in in the latest weeks. Only our internal beta. But the customer-visible version was last updated in 28/09. This problem started happening 08/10, minutes before I opened this issue.

The only thing we updated **today** was the OAuth backend: we had two HTTP 302 redirects after typing username/login. The first redirected to an "ok" page that then redirected back to an extension HTML file. This extension HTML file was the one that was causing the error above.

Even with a single `Office.onReady().then(() => console.log('ok')` in the script we were getting the error I posted above.

However cutting it down the login OAuth flow to one single redirect seems to have done the trick. I wonder if it's a timing issue, or maybe a browser security, and the Dialog lost communication with the Sidebar?

I was trying to make a minimal reproduction add-in. I will keep digging if something happens.

In the meantime I'll close this ticket, but I'll reopen if I can reproduce again.

Thanks for looking into it!
